fs/dynchunksize/dynchunksize.go

```diff
@@ -0,0 +1,37 @@
+// Dynamically calculate suitable chunk size for large uploads
+package dynchunksize
```

Comment:
I'm not a fan of this name!

Perhaps call the package `chunksize` and the function `Calculator`?